### PR TITLE
Build and push latest images on PR merge

### PR DESCRIPTION
**Pull request by PrasadG193**
## Change Overview

Build and push tools images with latest tag on every PR merge to master

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues

- #814

### CODE REVIEW

**Review by pavannd1 — COMMENTED** [review]
LGTM. @tdmanv @TribalNightOwl Please take a look once

**Review by TribalNightOwl — COMMENTED** [review]
LGTM but please wait for @tdmanv 's review

**Review by pavannd1 — APPROVED** [review]

File: .goreleaser.yml [hunk 1]
```diff
@@ -80,7 +80,7 @@ dockers:
   build_flag_templates:
   - "--build-arg=TOOLS_IMAGE=ghcr.io/kanisterio/kanister-tools:{{ .Tag }}"
 - image_templates:
-  - 'ghcr.io/kanisterio/kanister-kubectl:1.18'
+  - 'ghcr.io/kanisterio/kanister-kubectl:{{ .Tag }}'
```
[the remark this answers is not in the document]
Comment (PrasadG193) [reply .goreleaser.yml:83]: @tdmanv docker image tags should be immutable. Right now we are overriding the same image tag every time which is no correct. Ideally, the image tag should be updated on every change. So if there are any changes in source code which is changing `kanister-kubectl` image, we can simply update the imaged tag to something like `ghcr.io/kanisterio/kanister-kubectl:1.18-VERSION`

File: build/push_images.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+IMAGE_REGISTRY="ghcr.io/kanisterio"
+IMAGES=("mysql-sidecar" "kafka-adobe-s3-sink-connector" "postgres-kanister-tools" "postgresql" "postgres-tools-9.6" "cassandra" "kanister-kubectl:1.18" "mongo-sidecar" "mongodb" "es-sidecar" "controller" "kanister-tools" "couchbase-tools" "kafka-adobe-s3-source-connector" "foundationdb")
```
[the remark this answers is not in the document]
Comment (PrasadG193) [reply build/push_images.sh:21]: yes, thanks. Let me fix that

Comment (PrasadG193) [inline build/push_images.sh:21]:
```suggestion
IMAGES=("mysql-sidecar" "kafka-adobe-s3-sink-connector" "postgres-kanister-tools" "postgresql" "postgres-tools-9.6" "cassandra" "kanister-kubectl" "mongo-sidecar" "mongodb" "es-sidecar" "controller" "kanister-tools" "couchbase-tools" "kafka-adobe-s3-source-connector" "foundationdb")
```

File: .goreleaser.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrasadG193) [reply .goreleaser.yml:83]: Let's change the image tag format to `ghcr.io/kanisterio/kanister-kubectl:1.18-VERSION` or `ghcr.io/kanisterio/kanister-kubectl-1.18:VERSION`

[the remark this answers is not in the document]
Comment (tdmanv) [reply .goreleaser.yml:83]: Understood the need/desire for immutable, but we currently consume these images. Will that cause any issues?

[the remark this answers is not in the document]
Comment (PrasadG193) [reply .goreleaser.yml:83]: We won't be deleting the existing images from the registry, right? So it won't cause any issue. We will also update our existing blueprint to use images with newer format which we will keep updating on every release.

**Review by tdmanv — APPROVED** [review]
sounds good

File: .goreleaser.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (PrasadG193) [reply .goreleaser.yml]: Updated etcd blueprints to use `ghcr.io/kanisterio/kanister-kubectl-1.18:{{ .Tag }}` image tag

**Pull request merged**